gcp/app-engine/db_queries.py

```diff
@@ -0,0 +1,115 @@
+import sqlalchemy
+from datetime import datetime
+from db_utils import get_db_connection
+
+DB_SCHEMA = 'golfbetter'
+
+def get_next_tournaments(tournament_search=None):
+    query = f"""
+        select *
+        from
+        (
+          select name,
+            start_date,
+            is_completed,
+            data->>'date' as date,
+            data->>'city' as city,
+            coalesce(data->>'stateCode', '') as stateCode,
+            data->>'countryCode' as countryCode
+          from {DB_SCHEMA}.pga_tournaments
+          where ({f"lower(id) like '%{str(tournament_search).lower()}%' or lower(name) like '%{str(tournament_search).lower()}%'" if tournament_search is not None else "1=1"})
+          order by is_completed, abs(EXTRACT(EPOCH FROM (NOW() - start_date)))
+          limit {1 if tournament_search is not None else 5}
+        )
+        order by start_date
+    """
+    pool = get_db_connection()
+    now = datetime.now()
+    with pool.connect() as db_conn:
+        print(f"Running SQL query: {query}")
+        records = db_conn.execute(sqlalchemy.text(query)).fetchall()
+        tournaments = [{
+            "name": row[0],
+            "location": row[4] + (", " + row[5] + ' ' if row[5] else ' '),
+            "date": row[3],
+            "status": "Done" if row[2] else "Scheduled" if now.date() < row[1] else "In Progress"
+        } for row in records]
+        if len(tournaments) == 1:
+            return tournaments[0]
+        return tournaments
+
+def get_player(player_search, tournament_search, round=None):
```

Comment:
i made all of these files do a case-insensitive substring, so you can do something like "@Golf Better API player info thomps wyndh" to get player info about Thompson Davis at the Wyndham. I still need to clean up this quite a bit since it doesn't support spaces in the commands, etc., but is at least a start